src/Sylius/Component/Grid/Filter/MoneyFilter.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Grid\Filter;
+
+use Sylius\Component\Grid\Data\DataSourceInterface;
+use Sylius\Component\Grid\Filtering\FilterInterface;
+
+/**
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class MoneyFilter implements FilterInterface
+{
+    const NAME = 'money';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function apply(DataSourceInterface $dataSource, $name, $data, array $options)
+    {
+        if (empty($data) || empty($data['currency'])) {
+            return;
+        }
+
+        $field = isset($options['field']) ? $options['field'] : $name;
+
+        $greaterThan = $this->getDataValue($data, 'greaterThan');
+        $lessThan = $this->getDataValue($data, 'lessThan');
+
+        $expressionBuilder = $dataSource->getExpressionBuilder();
+
+        if ('' !== $greaterThan) {
+            $expressionBuilder->greaterThan($field, $this->normalizeAmount($greaterThan));
+        }
+        if ('' !== $lessThan) {
+            $expressionBuilder->lessThan($field, $this->normalizeAmount($lessThan));
+        }
+
+        $dataSource->restrict($expressionBuilder->equals($options['currencyField'], $data['currency']));
+    }
+
+    /**
+     * @param string|float $amount
+     *
+     * @return int
+     */
+    private function normalizeAmount($amount)
+    {
+        return (int) round($amount * 100);
```

Comment:
Maybe we could make this a bit more configurable and allow changing the multiplier in config? Just like for CurrencyType (symfony core form type). Default could be 100.